things/service/src/main/java/org/eclipse/ditto/things/service/persistence/actors/enrichment/ThingEventEnricher.java

```diff
@@ -0,0 +1,385 @@
+                                        .putHeader(extraFieldsKey,
+                                                buildPredefinedExtraFieldsHeaderList(combinedPredefinedExtraFields))
+                                        .putHeader(readGrantKey, indexedGrants.pathsToJson().toString())
+                                        .putHeader(extraFieldsObjectKey,
+                                                buildPredefinedExtraFieldsHeaderObject(thing,
+                                                        combinedPredefinedExtraFields).toString());
+                                
+                                return enrichedWithPartialAccessPaths.setDittoHeaders(headersBuilder.build());
+                            })
+            );
+        } else {
+            return partialAccessPathsStage;
+        }
+    }
+
+    private <T extends DittoHeadersSettable<? extends T>> CompletionStage<T> enrichWithPartialAccessPathsIfNecessary(
+            final T withDittoHeaders,
+            @Nullable final Thing thing,
+            @Nullable final PolicyId policyId) {
+
+        if (thing == null || !(withDittoHeaders instanceof ThingEvent<?> thingEvent)) {
+            return CompletableFuture.completedStage(withDittoHeaders);
+        }
+
+        return enrichWithPartialAccessPaths(thingEvent, thing, policyId)
+                .thenApply(partialAccessPathsJson -> {
+                    if (hasNoPartialSubjects(partialAccessPathsJson)) {
+                        return withDittoHeaders;
+                    }
+                    return addPartialAccessPathsHeader(withDittoHeaders, partialAccessPathsJson);
+                });
+    }
+
+    /**
+     * Applies RQL condition from config, using cached parsed criteria for performance.
+     * Caches only the Criteria (RQL AST), not the Predicate<Thing>, to allow building
+     * fresh predicates with current headers for each event.
+     */
+    private Predicate<PreDefinedExtraFieldsConfig> applyPredefinedExtraFieldsCondition(
+            final Thing thing,
+            final WithDittoHeaders withDittoHeaders
+    ) {
+        return conf -> {
+            final Optional<String> optCondition = conf.getCondition();
+            if (optCondition.isEmpty()) {
+                return true;
+            }
+
+            final String rqlCondition = optCondition.get();
+
+            try {
+                final Criteria criteria = RQL_CRITERIA_CACHE.computeIfAbsent(rqlCondition, rql -> {
+                    try {
+                        return QueryFilterCriteriaFactory
+                                .modelBased(RqlPredicateParser.getInstance())
+                                .filterCriteria(rql, DittoHeaders.empty());
+                    } catch (final InvalidRqlExpressionException e) {
+                        LOGGER.warn("Invalid RQL condition <{}> - ignoring: {}", rql, e.getMessage());
```

Comment:
Always provide the DittoHeaders to loggers, if they are present - that way e.g. the `correlation-id` and `trace-parent` are automatically added as MDC:
```
LOGGER.withCorrelationId(withDittoHeaders).warn(...
```

Ah, I see this was also missing before .. shame :D